build.js

```diff
@@ -0,0 +1,174 @@
+const StyleDictionary = require("style-dictionary").extend("config.json");
+
+const flattenObj = (key, obj) => {
+  if (obj === undefined) return;
+
+  const getEntries = () =>
+    Object.fromEntries(Object.entries(obj)?.map(([k, v]) => flattenObj(k, v)));
+
+  // We're checking for a 'value' key this way, rather than a simple obj.value
+  // check, because we want to avoid a false negative if obj.value resolves to a
+  // falsy value
+  const hasAValueKey = Object.keys(obj).includes("value");
+
+  return hasAValueKey ? [key, obj?.value] : [key, getEntries()];
+};
+
+const formatEntries = (entries) =>
+  !entries
+    ? ""
+    : JSON.stringify(
+        Object.fromEntries(
+          Object.entries(entries)?.map(([k, v]) => {
+            const isStringOrNum =
+              typeof v === "string" || typeof v === "number";
+            return isStringOrNum ? [k, v] : flattenObj(k, v);
+          })
+        )
+      );
+
+const parseFontWeight = (value) => {
+  if (typeof value !== "string") return value;
+  const val = value?.toLowerCase();
+  return val === "regular"
+    ? 400
+    : val === "medium"
+    ? 500
+    : val === "semi bold" || val === "semibold"
+    ? 600
+    : val === "bold"
+    ? 700
+    : val === "black"
+    ? 800
+    : val;
+};
```

Comment:
Ternaries get pretty hard to read when they aren't a single line expression. Can we convert this to a switch statement?

```suggestion
  const val = value?.toLowerCase().replace(" ", "");
  switch(val) {
    case "regular":
      return 400;
    cae "medium":
      return 500;
    case "semibold":
      return 600;
    case "bold":
      return 700;
    case "black":
      return 800;
    default:
      return val;
  }
};
```

Reply:
As a secondary note, this conversion of name to number feels unnecessary, and part of the contract between the font and CSS. Did you discover something new since we talked about this with Billy the other day?